src/constants.h

```diff
@@ -107,6 +107,15 @@ enum class Algorithm {
   kRs256Algorithm = -257
 };
 
+// Converts a the algorithm to a cbor::Value.
+cbor::Value CborValue(Algorithm alg);
+
+// Converts parameter, response and subcommand keys to cbor::Value.
+template <typename T>
+cbor::Value CborValue(T variant) {
+  return cbor::Value(static_cast<uint8_t>(variant));
+}
```

Comment:
This means that it is always possible to call `CborValue`. If we add a new type that should have its own implementation of `CborValue`, we would use this default if we forget to write an implementation. Not sure of the consequences.

Reply:
If there was a way to limit it to `enum class C : uint8_t`, I'd take it. Maybe rename it to CborU8 to make it more explicit? Most constants will be u8, so this is the most ergonomic solution.

Reply:
I don't know C++ enough to know if we can filter template inference to when something can be cast to `uint8_t` without loss of information.

Yes `CborU8` sounds like a good compromise. I'm also fine with the current solution. It just depends how often we expect to add a new CBOR-convertible type and how bad it may be to forget to implement its `CborValue` function.

Reply:
I expect to add some more parameter maps, with more u8.

I decided to go with `CborInt` as it constructs a CBOR Value of type Integer, and the static conversion to u8 is just an implementation detail of (most) of the conversions. The representation of the integers is not important for the function. Also that way you can't mistakenly call the wrong function for `Algorithm`.

Reply:
As I see it the initial problem is still present when adding a new integer-like type convertible to CBOR which doesn't convert loss-less to `uint8_t`. We still need to remember to add a custom overload. But maybe that's fine.